pallets/utxo/src/lib.rs

```diff
@@ -113,13 +113,23 @@ pub mod pallet {
     }
 
     impl TransactionInput {
-        pub fn new(outpoint: H256, sig_script: H512) -> Self {
+        /// New input with a signature in the `witness` field.
+        pub fn new_with_signature(outpoint: H256, sig_script: H512) -> Self {
             Self {
                 outpoint,
                 lock: Vec::new(),
                 witness: (&sig_script[..]).to_vec(),
             }
         }
+
+        /// New input with empty `lock` and `witness` to be filled later.
+        pub fn new_clean(outpoint: H256) -> Self {
```

Comment:
Is clean the best terminology here? Is new_empty better? Maybe it's just me

Reply:
Yea I was split between `empty` and `clean` and one other option that I don't remember. Naming things is a nightmare :). Will change that to `empty`.